src/coreclr/nativeaot/Runtime/wasm/wasi.c

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// This file exists to get access to the otherwise internal values of the __pthread.
+
+#include "wasi.h"
+#include <pthread.h>
+
+// copied from internal/pthread_impl.h
+#define __SU (sizeof(size_t)/sizeof(int))
+
+#define _a_stacksize __u.__s[0]
+#define _a_guardsize __u.__s[1]
+#define _a_stackaddr __u.__s[2]
+#define _a_detach __u.__i[3*__SU+0]
+
+struct __pthread {
```

Comment:
How much of this is needed (everything)?

Am I right to understand that we are basically polyfilling some pthread here because the underlying implementation doesn't provide them?

Reply:
Yes, Wasi_SDK's libc does not provide the non portable pthread methods, for us the one we want is `pthread_getattr_np` so I copied basically the emscripten version, but I need access to the private fields of the __pthread so I copied the definition of the struct and a few other macros from the internal file in the Wasi SDK.  It's all a bit fragile, I know.

Reply:
It is indeed fragile.

I dug around a bit and found these release notes:

https://github.com/WebAssembly/wasi-libc/blob/a6f871343313220b76009827ed0153586361c0d5/libc-top-half/musl/WHATSNEW#L873

Perhaps we could set `_GNU_SOURCE` and things would work?